### PR TITLE
Chapter 5: Adding Subtitles, Step in Indexing, and More Concise Solution.

### LINKED ISSUE #188: Syntax fix
In https://python-programming.quantecon.org/python_essentials.html, `%%file` should be in back ticks.

### PR DESCRIPTION
**Pull request by HumphreyYang**
Hi @jstac,

I have gone through the Python Essentials chapter and made the following changes:

1. added subtitles in the primitive type section to improve clarity;
2. changed %%file to %%writefile according to the [documentation for cell magic](https://ipython.readthedocs.io/en/stable/interactive/magics.html);
3. fixed #188 ;
4. added steps to the *Slice Notation* section;
5. added all() and any() function in the *Combining Expressions* section;
6. Wrote a one-line solution for exercise four using list comprehension;
7. Restructured exercise section #192 .

To keep this PR from being too heavy, I will open a separate PR for the migration of the *More on Function* section. A short section on *Naming Space* will also be added to *More on Function* after the migration. 

Could you please kindly review these changes and kindly provide me with feedback?

### CODE REVIEW

File: lectures/python_essentials.md [hunk 1]
```diff
@@ -940,14 +933,27 @@ Here's a solution:
 
 ```{code-cell} python3
```
Comment (HumphreyYang) [inline lectures/python_essentials.md:934]: To show my changes to the exercise solution clearly, here is the solution that I have rewritten.

File: lectures/python_essentials.md [hunk 2]
```diff
@@ -940,14 +933,27 @@ Here's a solution:
...
+```{code-cell} python3
+def f(seq_a, seq_b):
+  return all([i in seq_b for i in seq_a])
 
+  # == test == #
+print(f("ab", "cadb"))
+print(f("ab", "cjdb"))
 print(f([1, 2], [1, 2, 3]))
 print(f([1, 2, 3], [1, 2]))
 ```
```
Comment (HumphreyYang) [inline lectures/python_essentials.md]: To show my changes to the exercise solution clearly, here is where I have rewritten.

File: lectures/python_essentials.md [hunk 3]
```diff
@@ -61,6 +61,9 @@ Let's learn a bit more about them.
 
 ### Primitive Data Types
 
+
+***1. Boolean Values***
```
Comment (jstac) [inline lectures/python_essentials.md]: @HumphreyYang , please use markdown syntax for headings.  This matters for the latex/pdf version.

This should be 

`#### Boolean Values`

File: lectures/python_essentials.md [hunk 4]
```diff
@@ -207,6 +216,18 @@ Negative numbers are also permissible
 a[-2:]  # Last two elements of the list
 ```
 
+You can also use the format `[start:end:step]` to specify the step
+
+```{code-cell} python3
+a[::2]
+```
+
+Using the negative step, you can return the sequence in a reversed order
```
Comment (jstac) [inline lectures/python_essentials.md]: Using *a* negative step...

**Comment by jstac** [comment]
Great work @HumphreyYang , very nice.  Please see the minor comments above.

**Comment by HumphreyYang** [comment]
Hi @jstac,

Thank you so much for your comment.

I have changed the script based on your detailed comment. Could you please kindly merge this PR if you think it looks good to you?

**Pull request merged**